commit 00f2b0a9bb806be64c5868af44f43dbe55bdd75b
Author: Thomas Graf <tgraf@redhat.com>
Date:   Fri Sep 23 13:46:41 2011 +0200

    core: Include <linux/if.h> instead of <net/if.h>
    
    NM already includes <linux/if.h> in some places, f.e. nm-netlink-monitor and
    we can't mix usage of the two. Stick to using <linux/if.h> as it provides
    additional flag definitions such as operational link state and link mode.
    
    Signed-off-by: Thomas Graf <tgraf@redhat.com>
    
    [ Alexandre Rostovtsev <tetromino@gentoo.org>: remove parts of commit
    unrelated to if.h changes. ]

diff --git a/include/wireless-helper.h b/include/wireless-helper.h
index d150ef7..2e4509a 100644
--- a/include/wireless-helper.h
+++ b/include/wireless-helper.h
@@ -27,6 +27,6 @@
 #include <sys/types.h>
 #include <linux/types.h>
 #include <sys/socket.h>
-#include <net/if.h>
+#include <linux/if.h>
 #include <wireless.h>
 
diff --git a/src/nm-device.c b/src/nm-device.c
index c0b1b87..559606c 100644
--- a/src/nm-device.c
+++ b/src/nm-device.c
@@ -25,7 +25,6 @@
 #include <dbus/dbus.h>
 #include <netinet/in.h>
 #include <string.h>
-#include <net/if.h>
 #include <unistd.h>
 #include <errno.h>
 #include <sys/ioctl.h>
@@ -34,6 +33,7 @@
 #include <sys/wait.h>
 #include <arpa/inet.h>
 #include <fcntl.h>
+#include <linux/if.h>
 
 #include "nm-glib-compat.h"
 #include "nm-device-interface.h"
diff --git a/src/nm-system.c b/src/nm-system.c
index 0b29468..473fcec 100644
--- a/src/nm-system.c
+++ b/src/nm-system.c
@@ -40,7 +40,7 @@
 #include <netdb.h>
 #include <glib.h>
 #include <ctype.h>
-#include <net/if.h>
+#include <linux/if.h>
 
 #include "nm-system.h"
 #include "nm-device.h"
diff --git a/src/ppp-manager/nm-ppp-manager.c b/src/ppp-manager/nm-ppp-manager.c
index 522c075..3546f8d 100644
--- a/src/ppp-manager/nm-ppp-manager.c
+++ b/src/ppp-manager/nm-ppp-manager.c
@@ -32,7 +32,7 @@
 #include <sys/socket.h>
 #include <sys/ioctl.h>
 #include <asm/types.h>
-#include <net/if.h>
+#include <linux/if.h>
 #include <sys/stat.h>
 
 #include <linux/ppp_defs.h>
diff --git a/src/settings/plugins/ifcfg-rh/reader.c b/src/settings/plugins/ifcfg-rh/reader.c
index 691a176..910cca3 100644
--- a/src/settings/plugins/ifcfg-rh/reader.c
+++ b/src/settings/plugins/ifcfg-rh/reader.c
@@ -28,10 +28,10 @@
 #include <ctype.h>
 #include <sys/inotify.h>
 #include <errno.h>
-#include <net/if.h>
 #include <sys/ioctl.h>
 #include <unistd.h>
 #include <netinet/ether.h>
+#include <linux/if.h>
 
 #ifndef __user
 #define __user
diff --git a/src/wimax/iwmxsdk.c b/src/wimax/iwmxsdk.c
index ff6b553..9c3a78b 100644
--- a/src/wimax/iwmxsdk.c
+++ b/src/wimax/iwmxsdk.c
@@ -27,7 +27,8 @@
 #include <stdlib.h>
 #include <errno.h>
 #include <string.h>
-#include <net/if.h>
+#include <sys/socket.h>
+#include <linux/if.h>
 
 #include <glib.h>
 
diff --git a/src/wimax/nm-device-wimax.c b/src/wimax/nm-device-wimax.c
index b6afc27..6654140 100644
--- a/src/wimax/nm-device-wimax.c
+++ b/src/wimax/nm-device-wimax.c
@@ -23,7 +23,8 @@
 #include <unistd.h>
 #include <sys/ioctl.h>
 #include <net/ethernet.h>
-#include <net/if.h>
+#include <sys/socket.h>
+#include <linux/if.h>
 
 #include <WiMaxAPI.h>
 #include <WiMaxAPIEx.h>
